share/functions/__fish_complete_suffix.fish

```diff
@@ -8,71 +8,36 @@
 
-    # Variable declarations
+    argparse 'prefix=' 'desc=' 'comp=' -- $argv
 
-    set -l comp
-    set -l suff
-    set -l desc
-    set -l files
-    set -l prefix ""
-
-    switch (count $argv)
-
-        case 1
-            set comp (commandline -ct)
-            set suff $argv
-            set desc ""
-
-        case 2
-            set comp $argv[1]
-            set suff $argv[2]
-            set desc ""
-
-        case 3
-            set comp $argv[1]
-            set suff $argv[2]
-            set desc $argv[3]
-
-        case 4
-            set comp $argv[1]
-            set suff $argv[2]
-            set desc $argv[3]
-            set prefix $argv[4]
-
-            # Only directories are supported as prefixes, and to use the same logic
-            # for both absolute prefixed paths and relative non-prefixed paths, $prefix
-            # must terminate in a `/` if it is present, so it can be unconditionally
-            # prefixed to any path to get the desired result.
-            if not string match -qr '/$' $prefix
-                set prefix $prefix/
-            end
-    end
+    set -l suff (string escape --style=regex -- $argv)
 
     # Simple and common case: no prefix, just complete normally and sort matching files first.
-    if test -z $prefix
-        set -l suffix (string escape --style=regex -- $suff)
+    if test -z $_flag_prefix
         # Use normal file completions.
-        set files (complete -C "__fish_command_without_completions $comp")
-        set -l files_with_suffix (string match -r -- "^.*$suffix\$" $files)
+        set files (complete -C "__fish_command_without_completions $_flag_comp")
+        set -l files_with_suffix (string match -r -- (string join "|" "^.*"$suff\$) $files)
         set -l directories (string match -r -- '^.*/$' $files)
         set files $files_with_suffix $directories $files
     else
         # Strip leading ./ as it confuses the detection of base and suffix
         # It is conditionally re-added below.
-        set base $prefix(string replace -r '^("\')?\\./' '' -- $comp | string trim -c '\'"') # " make emacs syntax highlighting happy
+        set base $_flag_prefix(string replace -r '^("\')?\\./' '' -- $_flag_comp | string trim -c '\'"') # " make emacs syntax highlighting happy
 
         set -l all
         set -l files_with_suffix
         set -l dirs
-        # If $comp is "./ma" and the file is "main.py", we'll catch that case here,
+        # If $_flag_comp is "./ma" and the file is "main.py", we'll catch that case here,
         # but complete.cpp will not consider it a match, so we have to output the
         # correct form.
 
         # Also do directory completion, since there might be files with the correct
         # suffix in a subdirectory.
         set all $base*
-        set files_with_suffix (string match -r -- ".*"(string escape --style=regex -- $suff) $all)
+        set files_with_suffix (string match -r -- (string join "|" ".*"$suff) $all)
```

Comment:
I don't understand why it's ok to remove the regex escape here?

Reply:
```
# share/functions/__fish_complete_suffix.fish 
# line 6
 argparse 'prefix=' 'description=' 'complete=' -- $argv
```